### ISSUE TITLE
Not getting a warning when overlapping `stylelint-disable` ranges

### ISSUE THREAD

**Issue by kristerkari** [stylelint/stylelint#839]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

There is a bug with `stylelint-disable` CSS comment. It does not work when using it multiple times with nested selectors. When you disable something at the top level, it ignores the  `stylelint-disable` comments that follow.

> What CSS is needed to reproduce this issue?

```css
/* stylelint-disable any-rule */
.foo {
  /* stylelint-disable selector-no-id */
  .bar,
  #bar {
  /* stylelint-enable selector-no-id */
   color: black;
  }
}
```
The first `stylelint-disable` makes stylelint ignore the nested `stylelint-disable` comments.

> Which version of stylelint are you using?

`4.4.0`

> What did you expect to happen?

stylelint should not ignore the nested  `stylelint-disable` comments.

**Comment by davidtheclark** [comment]
You're right that this shouldn't be working. But you should be getting a warning. Are you not getting this warning: `A new disabled range cannot begin until the previous one has ended`? 

That's the way the disabling works right now. This could be a feature request to allow nested disable ranges.

**Comment by davidtheclark** [comment]
Also: Are you running stylelint before or after Sass or whatever else is resolving your nested selectors?

**Comment by kristerkari** [comment]
> But you should be getting a warning. Are you not getting this warning: A new disabled range cannot begin until the previous one has ended?

If I run stylelint using `grunt-postcss` or stylelint CLI, I'm not getting any warnings, except a linting error `Unexpected id selector (selector-no-id)`.

> Also: Are you running stylelint before or after Sass or whatever else is resolving your nested selectors?

I'm running it against a file that uses SCSS.

**Comment by kristerkari** [comment]
> This could be a feature request to allow nested disable ranges.

Should I create a new issue about it? Seems like a thing that is needed for anyone who uses nested selectors.

**Comment by kristerkari** [comment]
I also have to mention that Atom editor's `linter-stylelint` does not show any warnings, just a linting error `Unexpected id selector (selector-no-id)`

**Comment by kristerkari** [comment]
Another test case:

```css
// stylelint-disable x
.foo {
  /* stylelint-disable selector-no-id */
  .bar,
  #bar {
  /* stylelint-enable selector-no-id */
   color: black;
  }
}
```

**Comment by davidtheclark** [comment]
Yeah, sounds like it's not behaving as I thought it should. So I'll mark this one as a bug. If you could please file an issue for the feature request, that'd be great.

**Comment by kristerkari** [comment]
done: https://github.com/stylelint/stylelint/issues/840
